modules/exploits/windows/local/ms16_032_secondary_logon_handle_privesc.rb

```diff
@@ -0,0 +1,519 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'msf/core/payload_generator'
+require 'rex'
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = GoodRanking
+
+  include Msf::Exploit::Powershell
+  include Msf::Post::File
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'          => 'MS16-032 Secondary Logon Handle Privilege Escalation',
+      'Description'   => %q{
+This module exploits the lack of sanitization of standard handles in Windows' Secondary Logon Service.  The vulnerability is known to affect versions of Windows 7-10 and 2k8-2k12 32 and 64 bit.  This module will only work against those versions of Windows with Powershell 2.0 or later and systems with two or more CPU cores. Works about 75% of the time, if module fails, just re-run.
```

Comment:
If the module only works 75% of the time and its save to rerun, should it just retry X times until it succeeds?